IdentityCore/src/cache/ios/MSIDKeychainTokenCache.m

```diff
@@ -381,6 +381,59 @@ - (BOOL)saveAppMetadata:(MSIDAppMetadataCacheItem *)item
     return appMetadataitems;
 }
 
+- (BOOL)saveOrUpdateMetadataItem:(NSData *)item
+                          forKey:(MSIDCacheKey *)key
+                         context:(id<MSIDRequestContext>)context
+                           error:(NSError **)error
+{
+    if (!item)
+    {
+        if (error)
+        {
+            NSString *errorMessage = @"Nil metadata item is received!";
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, errorMessage, nil, nil, nil, context.correlationId, nil);
+        }
+        MSID_LOG_ERROR(context, @"Nil metadata item is received!");
+        return NO;
+    }
+    
+    MSID_LOG_VERBOSE(context, @"Saving metadata item info %@", item);
+    
+    return [self saveData:item
+                      key:key
+                  context:context
+                    error:error];
+}
+
+- (NSData *)metadataItemWithKey:(MSIDCacheKey *)key
+                        context:(id<MSIDRequestContext>)context
+                          error:(NSError **)error
+{
+    NSArray *items = [self itemsWithKey:key context:context error:error];
+    
+    if (!items || items.count < 1)
+    {
+        MSID_LOG_WARN(context, @"Found no metadata item.");
+        return nil;
+    }
+    
+    NSMutableArray *metadataitems = [NSMutableArray new];
+    
+    for (NSDictionary *attrs in items)
+    {
+        NSData *itemData = [attrs objectForKey:(id)kSecValueData];
+        
+        if (itemData)
+        {
+            [metadataitems addObject:itemData];
```

Comment:
why do you need to go through all of them if you only return the first one?

Reply:
There should only be one item per key, I will make a change to this to not create an array that we fill by checking all returned items. Instead, just return the one found.